Tasks/SqlDacpacDeploymentOnMachineGroup/ExecuteScriptsInTransaction.sql

```diff
@@ -0,0 +1,282 @@
+PRINT 'Begin Sql scripts execution'
+SET XACT_ABORT ON
+SET NOCOUNT ON
+
+DECLARE @_acquireLock VARCHAR(64)
+DECLARE @_acquireLockMilliseconds INT
+DECLARE @_acquireLockMaxAttempts INT
+DECLARE @_longRunningThresholdMilliSeconds INT
+DECLARE @_lastAttemptsToKillSessions INT
+DECLARE @_isHosted INT
+
+PRINT 'Adding SQL files into temporary table _tmpSqlFilesTable'
+
+IF OBJECT_ID (N'_tmpSqlFilesTable', N'U') IS NOT NULL 
+BEGIN
+   PRINT 'Temporary table exists already. Dropping table _tmpSqlFilesTable'
+   DROP TABLE _tmpSqlFilesTable
+END
+CREATE TABLE _tmpSqlFilesTable (ScriptIndex INT NOT NULL IDENTITY, SqlScriptContent NVARCHAR(MAX) NULL)
+
+IF OBJECT_ID (N'_vw_tmpSqlFilesTable', N'V') IS NOT NULL 
+BEGIN
+   PRINT 'View on temporary table exists already. Dropping view _vw_tmpSqlFilesTable'
+   DROP VIEW _vw_tmpSqlFilesTable
+END
+DECLARE @_viewCommand NVARCHAR(MAX)
+SET @_viewCommand = 'CREATE VIEW _vw_tmpSqlFilesTable AS SELECT t.SqlScriptContent from _tmpSqlFilesTable t'
+EXEC sp_executesql @_viewCommand
+
+DECLARE @_fileList NVARCHAR(MAX)
+DECLARE @_fileName NVARCHAR(MAX)
+DECLARE @_pos INT
+DECLARE @_insertFileCommand NVARCHAR(MAX)
+
+SET @_fileList = '$(_fileList)'
+WHILE LEN(@_fileList) > 0
+BEGIN
+	SET @_pos = CHARINDEX(';', @_fileList)
+	IF (@_pos = 0)
+	BEGIN
+		SET @_fileName = @_fileList
+		SET @_fileName = LTRIM(@_fileName)
+		SET @_fileName = RTRIM(@_fileName)
+		IF (LEN(@_fileName) > 0)
+		BEGIN
+			SET @_insertFileCommand = 'BULK INSERT _vw_tmpSqlFilesTable from ''' + @_fileName + ''' with (ROWTERMINATOR = ''\0'')'
+			EXEC sp_executesql @_insertFileCommand
+		END
+		SET @_fileList = ''
+	END
+	ELSE
+	BEGIN
+		SET @_fileName = SUBSTRING(@_fileList, 1, @_pos - 1)
+		SET @_fileName = LTRIM(@_fileName)
+		SET @_fileName = RTRIM(@_fileName)
+		IF (LEN(@_fileName) > 0)
+		BEGIN
+			SET @_insertFileCommand = 'BULK INSERT  _vw_tmpSqlFilesTable from ''' + @_fileName + ''' with (ROWTERMINATOR = ''\0'')'
+			EXEC sp_executesql @_insertFileCommand
+		END
```

Comment:
all these lines are common in both IF and ELSE blocks. These can be moved out leaving only @_fileName and @_fileList SET statements in blocks